From d78d17ad92012c92d7d0e492bdb9fd362d3cea06 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Thu, 16 Jan 2020 17:11:35 -0300
Subject: [PATCH] Fix setting focus on empty page

The first time the Activity is opened, there are no
pages in the GtkNotebook and therefore there is no
widget to set focus to.

We need to consider this scenario to avoid errors.
---
 notebook.py  | 5 ++++-
 pippy_app.py | 5 +++--
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/notebook.py b/notebook.py
index 52534f8..c229c32 100644
--- a/notebook.py
+++ b/notebook.py
@@ -317,7 +317,10 @@ class SourceNotebook(AddNotebook):
         return text_buffer
 
     def get_text_view(self):
-        tab = self.get_nth_page(self.get_current_page()).get_children()
+        page = self.get_current_page()
+        if page == -1:
+            return None
+        tab = self.get_nth_page(page).get_children()
         text_view = tab[0]
         return text_view
 
diff --git a/pippy_app.py b/pippy_app.py
index 65d91d8..5ffc4f2 100644
--- a/pippy_app.py
+++ b/pippy_app.py
@@ -166,8 +166,9 @@ class PippyActivity(ViewSourceActivity):
         def focus():
             """ Enforce focus for the text view once. """
             widget = self.get_toplevel().get_focus()
-            if widget is None:
-                self._source_tabs.get_text_view().grab_focus()
+            textview = self._source_tabs.get_text_view()
+            if widget is None and textview is not None:
+                textview.grab_focus()
                 return True
             return False
         GLib.timeout_add(100, focus)
-- 
2.23.0

